### PR TITLE
VZ-5822 Use Latest VMO and add VMO metrics e2e tests

### PR DESCRIPTION
**Pull request by zabdulre**
Updated bom to use the latest VMO version. Added tests to check the VMO metrics are available in prometheus. Modified network policy to allow prometheus scrape.

As the author of this PR, I have:

  

- [x] Checked that I included or updated copyright and license notices in all files that I altered

  

- [x] Added or updated unit tests for any new functions I added

  

- [x] Added or updated integration tests if appropriate

  

- [x] Added or updated acceptance tests if appropriate

Code reviewer, please confirm this PR:

  

- [ ] Addressed the requirement and meets the acceptance criteria

  

- [ ] Does not introduce unrelated or spurious changes

  

- [ ] Does not introduce any unapproved dependency

  

- [ ] Makes sense and it easy to understand, and/or difficult areas of code are clearly documented so that they can be understood

### CODE REVIEW

File: tests/e2e/metrics/syscomponents/metrics_test.go [hunk 1]
```diff
@@ -145,6 +150,23 @@ var _ = t.Describe("Prometheus Metrics", Label("f:observability.monitoring.prom"
 		t.ItMinimumVersion("Verify VPO upgrade counter metrics can be queried from Prometheus", "1.4.0", kubeConfig, func() {
 			eventuallyMetricsContainLabels("vz_nginx_upgrade_duration_seconds", map[string]string{})
 		})
+
+		t.ItMinimumVersion("Verify VMO function metrics can be queried from Prometheus", "1.4.0", kubeConfig, func() {
```
Comment (anders-swanson) [inline tests/e2e/metrics/syscomponents/metrics_test.go:154]: is there some contstant for 1.4.0 version

Comment (zabdulre) [reply tests/e2e/metrics/syscomponents/metrics_test.go:154]: that's true, let me do that now

File: tests/e2e/metrics/syscomponents/metrics_test.go [hunk 2]
```diff
@@ -365,6 +387,12 @@ func getClusterNameForPromQuery() string {
 	return ""
 }
 
+func eventuallyMetricsExistInCluster(metricName string, labels map[string]string, kconfig string) {
```
Comment (stevshap) [inline tests/e2e/metrics/syscomponents/metrics_test.go:390]: header comment would be nice

File: tests/e2e/metrics/syscomponents/metrics_test.go [hunk 3]
```diff
@@ -29,6 +29,10 @@ const (
 	sidecarInjectionRequests       = "sidecar_injection_requests_total"
 	prometheusTargetIntervalLength = "prometheus_target_interval_length_seconds"
 	envoyStatsRecentLookups        = "envoy_server_stats_recent_lookups"
+	vmoFunctionMetric              = "vmo_reconcile_total"
+	vmoCounterMetric               = "vmo_reconcile_total"
```
Comment (bobdonat) [inline tests/e2e/metrics/syscomponents/metrics_test.go:35]: is this correct? both metrics have the same name

File: tests/e2e/metrics/syscomponents/metrics_test.go (hunk 2 above)
Comment (zabdulre) [reply tests/e2e/metrics/syscomponents/metrics_test.go:392]: let me do that now

File: platform-operator/helm_config/charts/verrazzano-monitoring-operator/templates/networkpolicy.yaml [hunk 4]
```diff
@@ -26,3 +26,5 @@ spec:
       ports:
         - port: 15090
           protocol: TCP
+        - port: 9100
```
[the remark this answers is not in the document]
Comment (zabdulre) [reply platform-operator/helm_config/charts/verrazzano-monitoring-operator/templates/networkpolicy.yaml:29]: ok will do

File: tests/e2e/metrics/syscomponents/metrics_test.go (hunk 3 above)
Comment (zabdulre) [reply tests/e2e/metrics/syscomponents/metrics_test.go:35]: my mistake, thank you

File: platform-operator/verrazzano-bom.json [hunk 5]
```diff
@@ -701,4 +701,4 @@
       ]
     }
   ]
-}
+}
```
Comment (bobdonat) [inline platform-operator/verrazzano-bom.json:704]: nit newline was removed

File: tests/e2e/metrics/syscomponents/metrics_test.go [hunk 6]
```diff
@@ -365,6 +389,12 @@ func getClusterNameForPromQuery() string {
 	return ""
 }
 
+func eventuallyMetricsExistInCluster(metricName string, labels map[string]string, kconfig string) {
```
Comment (davisgomes) [inline tests/e2e/metrics/syscomponents/metrics_test.go:392]: Do we need a new function for this metrics check? Are you able to use the one George created below? I know they are internally a bit different, but I think they accomplish a similar function.

Comment (zabdulre) [reply tests/e2e/metrics/syscomponents/metrics_test.go:392]: Unfortunately not, the one George is using uses a query which does not work for the vmo metrics

**Review by davisgomes — APPROVED** [review]

**Pull request merged**